idaes/core/solvers/petsc.py

```diff
@@ -515,13 +538,39 @@ def petsc_dae_by_time_element(
                     options={"--ts_init_time": tprev, "--ts_max_time": t},
                 )
             if trajectory_save_prefix is not None:
+                tj_prev = tj
                 tj = PetscTrajectory(
                     stub=vars_stub, delete_on_read=True, unscale=t_block
                 )
-                tj.to_json(f"{trajectory_save_prefix}_{count}.json.gz")
+                if tj_prev is not None:
+                    # due to the way variables is generated we know varaibles
+                    # have corresponding positions in the list
+                    no_repeat = set()
+                    for i, v in enumerate(variables):
+                        vp = variables_prev[i]
+                        if id(v) in no_repeat:
+                            continue # variables can be repeated in list
+                        if isinstance(v.parent_component(), pyodae.DerivativeVar):
+                            continue # skip derivative vars
+                        no_repeat.add(id(v))
+                        # We'll add fixed vars in case they aren't fixed in
+                        # another section. Fixed vars don't go to the solver
+                        # so they don't show up in the trajectory data
+                        try:
+                            vec = tj.get_vec(v)
+                        except KeyError:
+                            vec = [pyo.value(v)]*len(tj.time)
+                        try:
+                            vec_prev = tj_prev.get_vec(vp)
+                        except KeyError:
+                            vec_prev = [pyo.value(vp)]*len(tj_prev.time)
+                        tj._set_vec(v, vec_prev + vec)
+                    tj._set_time_vec(tj_prev.time + tj.time)
+                variables_prev = variables
             tprev = t
-            count += 1
             res_list.append(res)
+        if tj is not None:
+            tj.to_json(f"{trajectory_save_prefix}_1.json.gz")
```

Comment:
Do we need the "_1" anymore?

Reply:
No, I just didn't want to break anyone's stuff.

Reply:
Better to make the change now, while there are only a few users.